neqo-transport/src/send_stream.rs

```diff
@@ -208,6 +216,22 @@ impl RangeTracker {
         let mut tmp_len = new_len;
         let mut v = Vec::new();
 
+        // Check for the common case of adding to the end
+        if let Some(mut last) = self.used.last_entry() {
+            let prev_off = *last.key();
+            let (prev_len, prev_state) = last.get_mut();
+            // allow for overlap between new chunk and the last entry
+            if new_off >= prev_off
+                && new_off <= prev_off + *prev_len
+                && new_off + new_len > prev_off + *prev_len
+                && new_state == *prev_state
```

Comment:
I'm sure that the compiler is capable of doing this, but maybe this check could be moved before the call to `.last_entry()`.

Reply:
We can't, it needs prev_state which comes from last_entry()